tests/fsharp/nunitConf.fs

```diff
@@ -0,0 +1,379 @@
+    let testContext () =
+        let test = NUnit.Framework.TestContext.CurrentContext.Test
+        { Directory = test.Properties.Get("DIRECTORY") :?> string;
+          Config = suiteHelpers.Value }
+
+// parametrized test cases does not inherits properties of test ( see https://github.com/nunit/nunit/issues/548 )
+// and properties is where the custom context data is saved
+
+type FSharpSuiteTestAttribute(dir: string) =
+    inherit NUnitAttribute()
+    interface NUnit.Framework.Interfaces.IApplyToTest with
+        member x.ApplyToTest(test: NUnit.Framework.Internal.Test) =
+            try
+                test.Properties |> FSharpTestSuite.setProps dir
+            with ex ->
+                test.RunState <- NUnit.Framework.Interfaces.RunState.NotRunnable
+                test.Properties.Set(NUnit.Framework.Internal.PropertyNames.SkipReason, NUnit.Framework.Internal.ExceptionHelper.BuildMessage(ex))
+                test.Properties.Set(NUnit.Framework.Internal.PropertyNames.ProviderStackTrace, NUnit.Framework.Internal.ExceptionHelper.BuildStackTrace(ex))
+
+type FSharpSuiteTestCaseData =
+    inherit TestCaseData
+
+    new (dir: string, [<ParamArray>] arguments: Object array) as this = 
+        { inherit TestCaseData(arguments) }
+        then
+            this.Properties |> FSharpTestSuite.setProps dir
+            arguments
+            |> Array.choose (fun a -> match a with :? Permutation as p -> Some p | _ -> None)
+            |> Array.iter (fun p -> this.SetCategory(sprintf "%A" p) |> ignore)
+
+[<AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)>]
+type FSharpSuiteTestCaseAttribute =
+    inherit TestCaseAttribute
+
+    new (dir: string, [<ParamArray>] arguments: Object array) as this = 
+        { inherit TestCaseAttribute(arguments) }
+        then
+            this.Properties |> FSharpTestSuite.setProps dir
+
+
+type FSharpSuitePermutationsAttribute(dir: string) =
+    inherit NUnitAttribute()
+
+    let _builder = NUnit.Framework.Internal.Builders.NUnitTestCaseBuilder()
+    interface NUnit.Framework.Interfaces.ITestBuilder with
+        member x.BuildFrom(methodInfo, suite) =
+            let allPermutations = 
+                [ FSI_FILE; FSI_STDIN; FSI_STDIN_OPT; FSI_STDIN_GUI;
+                  FSC_BASIC; FSC_HW; FSC_O3;
+                  GENERATED_SIGNATURE; EMPTY_SIGNATURE; EMPTY_SIGNATURE_OPT; 
+                  FSC_OPT_MINUS_DEBUG; FSC_OPT_PLUS_DEBUG; 
+                  FRENCH; SPANISH;
+                  AS_DLL; 
+                  WRAPPER_NAMESPACE; WRAPPER_NAMESPACE_OPT 
+                ]
+                |> List.map (fun p -> (new FSharpSuiteTestCaseData (dir, p)))
+
+            allPermutations
+            |> List.map (fun tc -> _builder.BuildTestMethod(methodInfo, suite, tc))
+            |> Seq.ofList
+
+module FileGuard =
+    let private remove path = if File.Exists(path) then Commands.rm (Path.GetTempPath()) path
+
+    [<AllowNullLiteral>]
+    type T (path: string) =
```

Comment:
Please don't use this non-idiomatic pattern.  Define FileGuard as a type (rather than a module)

    type FileGuard(path:string) = ...